Sources/Mock.swift

```diff
@@ -82,7 +82,19 @@ public struct Mock: Equatable {
     public var completion: (() -> Void)?
 
     /// The callback which will be executed everytime this `Mock` was started. Can be used within unit tests for validating that a request has been started. The callback must be set before calling `register`.
-    public var onRequest: OnRequest?
+    @available(*, deprecated, message: "Use `onRequestHandler` instead.")
+    public var onRequest: OnRequest? {
+        set {
+            onRequestHandler = OnRequestHandler(callback: newValue)
+        }
+        get {
+            assertionFailure("Getter is no longer supported")
```

Comment:
This will be even more breaking than it is right now; we should not deprecate but remove functionality.

We should continue supporting the older syntax, deprecate it, and remove support in a major version bump down the line.